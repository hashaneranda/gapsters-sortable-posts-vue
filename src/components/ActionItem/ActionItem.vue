<template>
  <div
    class="flex flex-row items-center max-w-full p-6 mx-auto mb-1 bg-white rounded-lg shadow-xl"
  >
    <h2 class="flex-grow leading-tight text-gray-900 text-l">{{ title }}</h2>

    <button
      class="p-2 ml-3 text-white transition duration-500 ease-in-out transform bg-green-400 rounded hover:bg-green-600 hover:-translate-y-1 hover:scale-110"
      v-on:click="() => handleTimeTravel()"
    >
      Time travel
    </button>
  </div>
</template>

<script>
export default {
  name: "ActionItem",
  props: {
    title: String,
    handleTimeTravel: Function,
  },
};
</script>
