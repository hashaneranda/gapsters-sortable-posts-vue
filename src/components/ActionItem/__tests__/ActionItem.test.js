import { render, cleanup, fireEvent } from "@testing-library/vue";
import { mount } from "@vue/test-utils";

import ActionItem from "../ActionItem";

afterEach(cleanup);

describe("ActionItem Componenet", () => {
  it("should take a snapshot", () => {
    const wrapper = mount(ActionItem);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should have a button", () => {
    const { getByTestId } = render(ActionItem);
    expect(getByTestId("button")).toHaveTextContent("Time travel");
  });

  it("renders componenet corretly without passing argumenets", () => {
    const { getByTestId } = render(ActionItem);
    expect(getByTestId("button")).toHaveTextContent("Time travel");
    expect(getByTestId("title")).toHaveTextContent("");
  });

  it("renders componenet corretly with title argumenet", () => {
    const { getByTestId } = render(ActionItem, {
      props: {
        title: "test title",
      },
    });
    expect(getByTestId("button")).toHaveTextContent("Time travel");
    expect(getByTestId("title")).toHaveTextContent("test title");
  });

  it("renders componenet corretly with title argumenet and handleTimeTravel argument", () => {
    const testFunction = () => {
      testVal = true;
    };

    const { getByTestId } = render(ActionItem, {
      props: {
        title: "test title",
        handleTimeTravel: testFunction,
      },
    });

    expect(getByTestId("button")).toHaveTextContent("Time travel");
    expect(getByTestId("title")).toHaveTextContent("test title");
  });

  it("function call working on button click", () => {
    let testVal = false;

    const testFunction = () => {
      testVal = true;
    };

    const { getByTestId } = render(ActionItem, {
      props: {
        title: "test title",
        handleTimeTravel: testFunction,
      },
    });
    fireEvent.click(getByTestId("button"));

    expect(getByTestId("button")).toHaveTextContent("Time travel");
    expect(getByTestId("title")).toHaveTextContent("test title");

    expect(testVal).toEqual(true);
  });
});
