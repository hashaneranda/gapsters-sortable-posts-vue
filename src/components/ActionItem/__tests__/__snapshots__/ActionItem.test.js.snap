// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionItem Componenet should take a snapshot 1`] = `
<div
  class="flex flex-row items-center max-w-full p-6 mx-auto mb-1 bg-white rounded-lg shadow-xl"
>
  <h2
    class="flex-grow leading-tight text-gray-900 text-l"
  >
    
  </h2>
   
  <button
    class="p-2 ml-3 text-white transition duration-500 ease-in-out transform bg-green-400 rounded hover:bg-green-600 hover:-translate-y-1 hover:scale-110"
  >
    
    Time travel
  
  </button>
</div>
`;
